src/settings.rs

```diff
@@ -496,17 +496,38 @@ impl Settings {
         settings
     }
 
+    // Use ~/.mcfly only if it already exists, otherwise create 'mcfly' folder in XDG_CACHE_DIR
     pub fn mcfly_training_cache_path() -> PathBuf {
-        Settings::storage_dir_path().join(PathBuf::from("training-cache.v1.csv"))
-    }
+        let cache_dir = Settings::mcfly_xdg_dir().cache_dir().to_path_buf();
 
-    pub fn storage_dir_path() -> PathBuf {
-        home_dir()
-            .unwrap_or_else(|| panic!("McFly error: Unable to access home directory"))
-            .join(PathBuf::from(".mcfly"))
+        Settings::mcfly_base_path(cache_dir).join(PathBuf::from("training-cache.v1.csv"))
     }
 
+    // Use ~/.mcfly only if it already exists, otherwise create 'mcfly' folder in XDG_DATA_DIR
     pub fn mcfly_db_path() -> PathBuf {
-        Settings::storage_dir_path().join(PathBuf::from("history.db"))
+        let data_dir = Settings::mcfly_xdg_dir().data_dir().to_path_buf();
+
+        Settings::mcfly_base_path(data_dir).join(PathBuf::from("history.db"))
+    }
+
+    fn mcfly_xdg_dir() -> ProjectDirs {
+        ProjectDirs::from("", "", "McFly").unwrap()
+    }
+
+    fn mcfly_base_path(base_dir: PathBuf) -> PathBuf {
+        let (home_exists, mcfly_path) = Settings::home_mcfly_exists();
+
+        if home_exists {
+            mcfly_path
+        } else {
+            base_dir
+        }
```

Comment:
If we return an option for `home_mcfly_exists` we can just us an `unwrap_or` here.